src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp

```diff
@@ -1499,6 +1499,37 @@ class ScheduleLogger {
         }
     }
 
+    void Schedule::applyWellProdIndexScaling(const std::string& well_name, const std::size_t reportStep, const double scalingFactor) {
+        auto wstat = this->wells_static.find(well_name);
+        if (wstat == this->wells_static.end())
+            return;
+
+        auto uwell = wstat->second.unique();
+        auto end   = uwell.end();
+        auto start = std::lower_bound(uwell.begin(), end, reportStep,
+            [](const auto& time_well_pair, const auto lookup) -> bool
+        {
+            //     time                 < reportStep
+            return time_well_pair.first < lookup;
+        });
+
+        if (start == end)
+            // Report step after last?
+            return;
+
+        // Relies on wells_static being OrderedMap<string, DynamicState<shared_ptr<>>>
+        // which means uwell is a vector<pair<time, shared_ptr<>>>
```

Comment:
Maybe rather `vector<std::size_t, shared_ptr<>>` - now one could think time was e.g. `std::time_t` or something similar?

Reply:
> Maybe rather `vector<std::size_t, shared_ptr<>>` - now one could think time was e.g. `std::time_t` or something similar?

I'm on the fence here.  The **value** is a time (a report step), but the **type** is obviously a `std::size_t`.  How about
```C++
vector<pair<report_step, shared_ptr<>>>
```
instead?

Reply:
I like the alternative with `report_step`

Reply:
> I like the alternative with `report_step`

I've updated the comment accordingly.